modules/mod_ginger_spa/controllers/controller_ginger_spa_template.erl

```diff
@@ -0,0 +1,83 @@
+%% @copyright Driebit BV 2021
+%% @doc Controller to render templates. Post arguments via the query arguments.
+
+-module(controller_ginger_spa_template).
+
+-export([init/1, service_available/2, charsets_provided/2, content_types_provided/2]).
+-export([allowed_methods/2, resource_exists/2]).
+-export([process_post/2, provide_content/2]).
+
+-include_lib("controller_webmachine_helper.hrl").
+-include_lib("include/zotonic.hrl").
+
+init(DispatchArgs) -> {ok, DispatchArgs}.
+
+service_available(ReqData, DispatchArgs) when is_list(DispatchArgs) ->
+    Context1 = z_context:new_request(ReqData, DispatchArgs, ?MODULE),
+    ?WM_REPLY(true, Context1).
+
+allowed_methods(ReqData, Context) ->
+    {[ 'POST', 'GET', 'HEAD' ], ReqData, Context}.
+
+charsets_provided(ReqData, Context) ->
+    {[{"utf-8", fun(X) -> X end}], ReqData, Context}.
+
+content_types_provided(ReqData, Context) ->
+    {[{"text/plain", provide_content}], ReqData, Context}.
+
+resource_exists(ReqData, Context) ->
+    Context1 = ?WM_REQ(ReqData, Context),
+    Context2 = z_context:ensure_all(Context1),
+    ?WM_REPLY(true, Context2).
+
+process_post(ReqData, Context) ->
+    Context1 = ?WM_REQ(ReqData, Context),
```

Comment:
Is this strictly necessary? Would returning Context2 in resource_exists (which is visited before process_post) not already have the ReqData WM_REQ'ed in?